filebeat/harvester/source/source.go

```diff
@@ -0,0 +1,17 @@
+package source
+
+import (
+	"io"
+	"os"
+)
+
+type LogSource interface {
+	io.ReadCloser
+	Name() string
+}
+
+type FileSource interface {
+	LogSource
+	Stat() (os.FileInfo, error)
+	Continuable() bool // can we continue processing after EOF?
```

Comment:
Same reason, pipeSource (now source.Pipe) exists. In order to massage `stdin` to work well `harvester/log` it's reader and truncation checks without generating panics.